facet-json/src/deserialize.rs

```diff
@@ -54,40 +56,54 @@ pub enum JsonErrorKind {
 
 impl core::fmt::Display for JsonParseErrorWithContext<'_> {
     fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
-        let Ok(input) = core::str::from_utf8(self.input) else {
+        let Ok(input_str) = core::str::from_utf8(self.input) else {
             return write!(f, "(JSON input was invalid UTF-8)");
         };
 
-        writeln!(f)?;
-        writeln!(f, "Error at {}:", self.path.clone().bright_yellow())?;
-
-        let mut pos = self.pos as isize;
-        let mut line_start = 0;
-        for (line_num, line) in input.lines().enumerate() {
-            let line_len = line.len();
-            let pos_line = pos >= 0 && pos <= line_len as isize;
-            let dist = (pos - line_start).abs();
-
-            if dist < 120 || pos_line {
-                writeln!(f, "{:>6} {}", (line_num + 1).dim(), line)?;
-                if pos_line {
-                    writeln!(
-                        f,
-                        "{:width$} {} {:?}",
-                        "",
-                        "⎩".red(),
-                        self.kind.bright_blue(),
-                        width = (pos as usize + 6), // 6 for line numbers, 1 for spaces
-                    )?;
-                }
-            } else if dist > 180 && pos < 0 {
-                break;
+        let source_id = "json";
+
+        let (span_start, span_end) = match &self.kind {
+            JsonErrorKind::StringAsNumber(s) => (self.pos - s.len(), self.pos),
+            _ => {
+                let span_end = if self.pos < self.input.len() {
+                    self.pos + 1
+                } else {
+                    self.input.len()
+                };
+                (self.pos, span_end)
             }
+        };
+
+        let mut report = Report::build(ReportKind::Error, (source_id, span_start..span_end))
+            .with_message(format!("Error at {}", self.path.fg(Color::Yellow)));
```

Comment:
You might want to add `.with_config(Config::new().with_index_type(IndexType::Byte))`.

Ariadne currently defaults to character indices: this is a misstep from the past that's going to be fixed soon.

Reply:
Ah wow thanks for reviewing! Opened #257 to track this :slightly_smiling_face: